include/measurement_kit/internal/vendor/mkcollector.hpp

```diff
@@ -21,8 +21,9 @@ class Settings {
   /// ca_bundle_path is the path to the CA bundle (required on mobile)
   std::string ca_bundle_path;
 
-  /// timeout is the whole operation timeout (in seconds)
-  int64_t timeout = 30;
+  /// timeout is the whole operation timeout (in seconds). Zero indicatest
+  /// that there actually is no timeout.
+  int64_t timeout = 0;
```

Comment:
so we leave the timeout managment entirely to the app?

Reply:
Yes. In all the cases where a default timeout was a bad idea, I zeroed the timeout. Which means that, if the app sets no timeout, there's no timeout!

Reply:
What is the implication of a no timeout? the operation can run forever?

Reply:
Yes, it's up to the app to set a timeout. I think, if the app does not set one, the user would eventually be bored and kill the app. I'm designing a better scheme in Go where the user would be able to interrupt a pending operation. Theoretically, I could also implement it in C++, but I think it's probably best to focus on Go at this point, as it's definitely easier.

Reply:
Ok I'm gonna check that actually there is a timeout everywhere in the app and approve this PR